### ISSUE TITLE
Test with file in subdirectory fail

### ISSUE THREAD

**Issue by petterreinholdtsen** [unioslo/tsdfx#98]
Based on a observation in pull request #95, I decided to try to write a test with a file in a subdirectory, but am unable to get it working.

I used this patch.

```
diff --git a/t/test-simplecopy.sh b/t/test-simplecopy.sh
index 2c2bfa0..8c9d249 100755
--- a/t/test-simplecopy.sh
+++ b/t/test-simplecopy.sh
@@ -13,6 +13,8 @@ echo "badfile, should be ignored" > "${srcdir}/$(printf "\002")"
 echo test1 > "${srcdir}/test1"
 echo 'test(2)' > "${srcdir}/test (2).txt"
 echo test2 > "${srcdir}/test2"
+mkdir "${srcdir}/d"
+echo test3 > "${srcdir}/d/test3"

 dd bs=1k count=${randomsize} \
     if=/dev/urandom \
@@ -21,8 +23,10 @@ dd bs=1k count=${randomsize} \
 md5start=$(cd ${srcdir}; md5sum ${randomsize}krandom)

 run_daemon -1
+run_daemon -1
+run_daemon -1

-for good in test1 test2 ; do
+for good in test1 test2 d/test3; do
        if [ ! -e "${dstdir}/${good}" ] ; then
                fail_test "missing: ${dstdir}/${good}"
        elif ! cmp -s "${srcdir}/${good}" "${dstdir}/${good}" ; then
```

I expected the first daemon run to create the directory and the second to copy the file, but this is not happening.  Anyone got an idea why?

**Comment by petterreinholdtsen** [comment]
Hm, when I test this code again today, it work.   Wierd.

**Comment by petterreinholdtsen** [comment]
I added this test to the master branch and included it in version v1.20160621.  Building the code failed because the new test-simplecopy.sh check now failed.  I guess there must be a race somewhere in the code.

**Comment by petterreinholdtsen** [comment]
The production build failed first, and then succeeded on the second try, without any code changes.  Definitely a race in the test code.  Will have to back out the test until it become stable.

**Comment by petterreinholdtsen** [comment]
Fixed with pull request #102 .
